lib/msal-browser/src/broker/nativeBroker/PlatformAuthDOMHandler.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Logger,
+    createAuthError,
+    AuthErrorCodes,
+    IPerformanceClient,
+    StringDict,
+} from "@azure/msal-common/browser";
+import {
+    PlatformBrokerRequest,
+    PlatformDOMTokenRequest,
+} from "./PlatformBrokerRequest.js";
+import { createNewGuid } from "../../crypto/BrowserCrypto.js";
+import { NativeConstants } from "../../utils/BrowserConstants.js";
+import {
+    PlatformBrokerResponse,
+    PlatformDOMTokenResponse,
+} from "./PlatformBrokerResponse.js";
+import { createNativeAuthError } from "../../error/NativeAuthError.js";
+import { IPlatformAuthHandler } from "./IPlatformAuthHandler.js";
+
+export class PlatformAuthDOMHandler implements IPlatformAuthHandler {
+    protected logger: Logger;
+    protected performanceClient: IPerformanceClient;
+    protected correlationId: string;
+    protected brokerId: string;
+    protected extensionVersion: string;
+    platformAuthType: string;
+
+    constructor(
+        logger: Logger,
+        performanceClient: IPerformanceClient,
+        brokerId?: string,
+        correlationId?: string
+    ) {
+        this.logger = logger;
+        this.performanceClient = performanceClient;
+        this.brokerId = brokerId || NativeConstants.MICROSOFT_ENTRA_BROKERID;
+        this.correlationId = correlationId || createNewGuid();
+        this.extensionVersion = "";
+        this.platformAuthType = NativeConstants.PLATFORM_DOM_PROVIDER;
+    }
+
+    /**
+     * Returns the Id for the broker extension this handler is communicating with
+     * @returns
+     */
+    getExtensionId(): string {
+        return this.brokerId;
+    }
+
+    /**
+     * Gets the version of the browser this handler is communicating with
+     */
+    getExtensionVersion(): string | undefined {
+        // @ts-ignore
+        const userAgent = window.navigator.userAgentData?.getHighEntropyValues([
+            "uaFullVersion",
```

Comment:
Looks like this is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues

Reply:
And also returns a promise